sudoku_board.cc

```diff
@@ -0,0 +1,123 @@
+
+#include "sudoku_board.h"
+
+// ------------------------------- public methods ---------------------------------
+
+SudokuBoard::SudokuBoard(int size){
+    m_size = size;
+    m_board_values = std::vector(size, std::vector<int>(size, EMPTY_CELL_SIGN));
+    m_possible_options = std::vector(size, std::vector(size, std::set<int>(POSSIBLE_CELL_VALUES)));
+}
+
+SudokuBoard::SudokuBoard(int size, std::vector<std::vector<int>> board_values){
+    m_size = size;
+    m_board_values = board_values; //question: when to use a copy constructor?
```

Comment:
Here you used (implicitly) a copy assignment - copy assignment happens when you copy a value into another **existing** value. Copy constructor happens when you create a copy of a value.
Note here that you first copy (from the outside) `board_values` and then you copy it again into `m_board_values`.
A better solution here would be to move `board_values` into `m_board_values`, and thereby just make `m_board_values` steal the resources from `board_values`.
Also, use the initialization list.